pandas/tests/frame/test_constructors.py

```diff
@@ -1340,6 +1340,31 @@ def test_constructor_list_of_namedtuples(self):
         result = DataFrame(tuples, columns=["y", "z"])
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.skipif(not PY36, reason="Guaranteed dict order is Python>=3.7")
+    def test_constructor_list_of_dict_order(self):
+        # GH10056
+        data = [
+            {"First": 1, "Second": 4, "Third": 7, "Fourth": 10},
+            {"Second": 5, "First": 2, "Fourth": 11, "Third": 8},
+            {"Second": 6, "First": 3, "Fourth": 12, "Third": 9, "YYY": 14, "XXX": 13},
+        ]
+        expected = DataFrame(
+            {
+                "First": [1, 2, 3],
+                "Second": [4, 5, 6],
+                "Third": [7, 8, 9],
+                "Fourth": [10, 11, 12],
+                "XXX": [None, None, 13],
+                "YYY": [None, None, 14],
+            }
+        )
+        result = DataFrame(data)
+        assert set(result.columns) == set(expected.columns)
+        # order of first 4 columns dictated by data[0]
+        tm.assert_frame_equal(result.iloc[:, :4], expected.iloc[:, :4])
+        # the order of the remaining two is arbitrary, but assert the data matches
```

Comment:
Did you actually observe that this was arbitrary? 
(I already tried to explain it is not)

Reply:
it was in with my original solution, it's not now but I forgot to update it (and it passed). 

Pushed a fix.